springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/payload/internal/PayloadService.java

```diff
@@ -49,4 +56,30 @@ public NamedSchemaObject useUnusedPayload() {
         }
         return PAYLOAD_NOT_USED;
     }
+
+    private SchemaObject composePolymorphicSchema(String schemaName, Class<?> clazz) {
+        List<MessageReference> messageReferences = componentsService.getSchemas().entrySet().stream()
+                .filter(e -> e.getValue().getAllOf() != null)
+                .filter(e -> e.getValue().getAllOf().stream()
+                        .anyMatch(s2 -> MessageReference.toSchema(schemaName).equals(s2.getReference())))
+                .map(e -> MessageReference.toSchema(e.getKey()))
+                .toList();
+        if (messageReferences.isEmpty()) {
+            return null;
+        }
+        if (isConcreteClass(clazz)) {
+            messageReferences = new ArrayList<>(messageReferences);
+            messageReferences.add(MessageReference.toSchema(schemaName));
+        }
+        return SchemaObject.builder()
+                .oneOf(messageReferences.stream()
+                        .sorted(Comparator.comparing(MessageReference::getRef))
```

Comment:
side note: Can the ObjectMapper sort the entries at the end?

In earlier versions, we tried sorting entries manually, but over time it became more and more difficult to maintain and debug.